public/js/cin.js

```diff
@@ -0,0 +1,527 @@
+            >
+          </div>
+        </div>
+      </div>
+    `;
+  }
+
+  return html`
+    <div>
+      <h1 class="display-3 cin-text text-center">All Interviews</h1>
+      <div class="row">
+        <div class="col">${slots}</div>
+        ${youtubeColumn}
+      </div>
+    </div>
+  `;
+};
+
+const LiveArea = () => {
+  const [keepAlive, setKeepAlive] = useState(false);
+  const [positions, setPositions] = useState([
+    "Live Position",
+    "Next Position",
+  ]);
+  const [candidates, setCandidates] = useState([
+    "Live Candidate",
+    "Next Candidate",
+  ]);
+  const [interviewers, setInterviewers] = useState([
+    "Live Interviewer",
+    "Next Interviewer",
+  ]);
+  const [times, setTimes] = useState(["Live Time", "Next Time"]);
+  const [showLive, setShowLive] = useState(false);
+  const [showNext, setShowNext] = useState(false);
+
+  const updateLives = async () => {
+    setKeepAlive(!keepAlive);
+    console.log("Updating Live Tiles");
+    for (let i = 0; i < interviews.length; i++) {
+      if (
+        new Date(interviews[i].start_time).getTime() < Date.now() &&
+        new Date(interviews[i].end_time).getTime() > Date.now()
+      ) {
+        setShowLive(true);
+        if (i + 1 != interviews.length) {
+          setShowNext(true);
+          setPositions([
+            interviews[i].interview.position.full_name,
+            interviews[i + 1].interview.position.full_name,
+          ]);
+
+          setCandidates([
+            prettifyCandidates(interviews[i].interview.candidates),
+            prettifyCandidates(interviews[i + 1].interview.candidates),
+          ]);
+
+          setInterviewers([
+            getInterviewers(interviews[i]),
+            getInterviewers(interviews[i + 1]),
+          ]);
+
+          setTimes([
+            "Now - " +
+              new Date(interviews[i].end_time)
+                .toLocaleTimeString()
+                .slice(0, -3),
+            new Date(interviews[i + 1].start_time)
+              .toLocaleTimeString()
+              .slice(0, -3) +
+              " - " +
+              new Date(interviews[i + 1].end_time)
+                .toLocaleTimeString()
+                .slice(0, -3),
+          ]);
+          break;
+        } else {
+          setShowNext(false);
+          setPositions([interviews[i].interview.position.full_name, ""]);
+          setCandidates([
+            prettifyCandidates(interviews[i].interview.candidates),
+            "",
+          ]);
+          setInterviewers([getInterviewers(interviews[i]), ""]);
+          setTimes([
+            "Now - " +
+              new Date(interviews[i].end_time)
+                .toLocaleTimeString()
+                .slice(0, -3),
+            "",
+          ]);
+          break;
+        }
+      } else {
+        setShowLive(false);
+        var intDate = new Date(interviews[i].start_time);
+        if (
+          intDate.getTime() > Date.now() &&
+          intDate.getTime() < Date.now() + 900000
```

Comment:
?

Reply:
if we're in a green room segment, there won't be anything live, so we just have a next up. but we don't want it saying "next up" all week leading up to cin, so i've just gone for "next up within 15 mins"